README.md

```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
 flag) otherwise they fallback to using main thread rAF to emulate the behaviour.
 
 
+# Key Concepts
+
+## WrokletAnimation
+`WorkletAnimation` is a subclass of Animation that can be used to create an custom animation effect
+that runs inside the animation worklet scope. A worklet animation has a  corresponding animator
+instance in the animation worklet scope which is responsible to drive its keyframe effects.
+Here are the key differences compared to a regular web animation:
+  - AnimationId should match a specific animatior class registered in worklet scope.
+  - WorkletAnimation may have multiple timelines (including ScrollTimelines).
+  - WorkletAnimation may have a custom properties bag that can be cloned and provided to animator
+    constructor when it is being instantiated.
+
+Note that worklet animations expose same API aurface as other web animations and thus they may be
+created, played, paused, inspected, and generally controlled from main document scope. Here is how
+various methods roughly translate:
+
+  - cancel(): cancels the animation and the corresponding animator instance is removed.
+  - play(): starts the animation and the corresponding animator instance may get its `animate` function
+    called periodically as a result of changes in its timelines.
+  - pause(): pauses the animation and the corresponding animator instance no longer receives
+    `animate` calls.
+  - finish(): invokes `finish` on the corresponding animator instance.
+  - reverse() or mutating playbackRate: invokes `playbackRateChanged` on the corresponding
+     animator instance.
+
+## ScrollTimeline
+[ScrollTimline](https://wicg.github.io/scroll-animations/#scrolltimeline) is a concept introduced in
+scroll-linked animation proposal. It defines an animation timeline whose time value depends on
+scroll position of a scroll container. ScrollTimeline can be used an an input timeline for
+worklet animations and it is the intended mechanisms to give read access to scroll position.
+
+## GroupEffect
+[GroupEffect](https://w3c.github.io/web-animations/level-2/#the-animationgroup-interfaces) is a
+concept introduced in Web Animation Level 2 specification. It provides a way to group multiple
+effects in a tree structure. GroupEffect can be used as the output for worklet animations. It
+makes it possible for worklet animation to drive effects spanning multiple elements.
+
+**TODO**: At the moment, group effect define strict rules to translate group effect time to children
+effect times but it given flexibility of children start time. This is not flexible enough for
```

Comment:
Explained in more details what current limitation of GroupEffect is and what the flexibility we want to add translates to.